src/ensembl/src/content/app/entity-viewer/shared/rest/rest-data-fetchers/proteinData.ts

```diff
@@ -0,0 +1,48 @@
+import { restProteinStatsAdaptor } from '../rest-adaptors/rest-protein-adaptor';
+import { TranscriptInResponse } from './transcriptData';
+
+export type XrefsInResponse = {
+  display_id: string;
+}[];
+
+export type ProteinStatsInResponse = {
+  pdbs: number;
+  ligands: number;
+  interaction_partners: number;
+  annotations: number;
+};
+
+export type UniProtSummaryStats = {
+  [key: string]: ProteinStatsInResponse;
+};
+
+export type ProteinStats = Pick<
+  ProteinStatsInResponse,
+  'ligands' | 'annotations'
+> & {
+  structures: number;
+  interactions: number;
+};
+
+export const fetchProteinSummaryStats = async (
+  transcriptId: string,
+  signal?: AbortSignal
+): Promise<ProteinStats> => {
+  const transcriptUrl = `http://rest.ensembl.org/lookup/id/${transcriptId}?expand=1;content-type=application/json`;
+  const transcript: TranscriptInResponse = (await fetch(transcriptUrl, {
+    signal
+  }).then((response) => response.json())) as TranscriptInResponse;
+
+  const xrefsUrl = `http://rest.ensembl.org/xrefs/id/${transcript.Translation?.id}?content-type=application/json;external_db=Uniprot/SWISSPROT`;
+  const xrefsData: XrefsInResponse = (await fetch(xrefsUrl, {
+    signal
+  }).then((response) => response.json())) as XrefsInResponse;
+  const pdbeId = xrefsData[0].display_id;
```

Comment:
`xrefsData[0]` can be undefined

e.g.:  http://rest.ensembl.org/xrefs/id/ENSP00000499308?content-type=application/json;external_db=Uniprot/SWISSPROT

Also, I expect all XHR requests to http urls will fail once you deploy the site to a url with https. Use https instead of http in links.

Reply:
✔️